### PR TITLE
Improve suggestion provider

### LINKED ISSUE #517: Add simple suggestion provider
On app start/login (and maybe after finishing a pull sync, especially when logging initially) we want to show simple suggestions: Basically, go over all time entries of the previous 4 weeks, and suggest the most common (description, project(, task)).

@pe0ny How do tags work with suggestions if at all?

### PR DESCRIPTION
**Pull request by willsb**
Closes #517 

Squash whenever

### CODE REVIEW

File: Toggl.Foundation.Tests/Suggestions/MostUsedTimeEntrySuggestionProviderTests.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+                suggestions.Should().HaveCount(5);
+            }
+
+            [Fact]
+            public async Task SortsTheSuggestionsByUsage()
+            {
+                var suggestions = await Provider.GetSuggestions().ToList();
+
+                suggestions.FirstOrDefault(s => s.Description == "te2").Should().BeNull();
```
Comment (simonrozsival) [inline Toggl.Foundation.Tests/Suggestions/MostUsedTimeEntrySuggestionProviderTests.cs]: This tests only that the "te2", which is the least used one, is not returned in the list of suggestions. Why don't you test the exact order of the returned list of suggestions?

File: Toggl.Foundation/Suggestions/MostUsedTimeEntrySuggestionProvider.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+using Toggl.PrimeRadiant;
+using Toggl.PrimeRadiant.Models;
+
+namespace Toggl.Foundation.Suggestions
+{
+    public sealed class MostUsedTimeEntrySuggestionProvider : ISuggestionProvider
+    {
+        private const int daysToQuery = 42;
+        private const int maxNumberOfSuggestions = 5;
+        private readonly TimeSpan fourWeeks = TimeSpan.FromDays(daysToQuery);
```
Comment (simonrozsival) [inline Toggl.Foundation/Suggestions/MostUsedTimeEntrySuggestionProvider.cs]: Please change the name of the field - it is not four weeks, but six weeks. Ideally change to something value-agnostic.

File: Toggl.Foundation/Suggestions/MostUsedTimeEntrySuggestionProvider.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+        private readonly ITogglDatabase database;
+        private readonly ITimeService timeService;
+
+        public MostUsedTimeEntrySuggestionProvider(ITogglDatabase database, ITimeService timeService)
```
Comment (simonrozsival) [inline Toggl.Foundation/Suggestions/MostUsedTimeEntrySuggestionProvider.cs]: Maybe you could inject the repository directly here.

File: Toggl.Foundation/Suggestions/MostUsedTimeEntrySuggestionProvider.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            Ensure.Argument.IsNotNull(database, nameof(database));
+            Ensure.Argument.IsNotNull(timeService, nameof(timeService));
+
+            this.database = database;
+            this.timeService = timeService;
+        }
+
+        public IObservable<Suggestion> GetSuggestions()
+            => database.TimeEntries
+                       .GetAll(isInLastTwoWeeks)
```
Comment (simonrozsival) [inline Toggl.Foundation/Suggestions/MostUsedTimeEntrySuggestionProvider.cs]: 2 weeks? 4 weeks? 6 weeks? :-D Please rename.

File: Toggl.Foundation/Suggestions/MostUsedTimeEntrySuggestionProvider.cs (hunk 3 above)
Comment (willsb) [reply Toggl.Foundation/Suggestions/MostUsedTimeEntrySuggestionProvider.cs]: Maybe, but what's the benefit?

File: Toggl.Foundation/Suggestions/MostUsedTimeEntrySuggestionProvider.cs (hunk 4 above)
Comment (willsb) [reply Toggl.Foundation/Suggestions/MostUsedTimeEntrySuggestionProvider.cs]: `isInLastFewWeeks` 😄

File: Toggl.Foundation/Suggestions/MostUsedTimeEntrySuggestionProvider.cs (hunk 3 above)
Comment (simonrozsival) [reply Toggl.Foundation/Suggestions/MostUsedTimeEntrySuggestionProvider.cs]: This is merely a suggestion. Feel free to dismiss it.

File: Toggl.Foundation.Tests/Suggestions/MostUsedTimeEntrySuggestionProviderTests.cs [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+                            createTimeEntry("te1", 1, 1),
+                            createTimeEntry("te1", 1, 1),
+                            createTimeEntry("te1", 1, 1),
+                            createTimeEntry("te2", 1, 2),
+                            createTimeEntry("te3", 1, 1),
+                            createTimeEntry("te3", 1, 1),
+                            createTimeEntry("te4", 1, 2),
+                            createTimeEntry("te4", 1, 2),
+                            createTimeEntry("te5", 2, 2),
+                            createTimeEntry("te5", 2, 2)
```
Comment (arturs105) [inline Toggl.Foundation.Tests/Suggestions/MostUsedTimeEntrySuggestionProviderTests.cs]: Could we change this to a property test? Might be some problems with values being unique (not repeated), but shouldn't be too hard to solve that, however, is it worth the effort?

**Pull request merged**